lib/vcr/middleware/net_http2.rb

```diff
@@ -0,0 +1,86 @@
+require 'net-http2'
+require 'vcr/util/version_checker'
+require 'vcr/request_handler'
+
+VCR::VersionChecker.new('NetHttp2', NetHttp2::VERSION, '0.7.0').check_version!
+
+module VCR
+  # Contains middlewares for use with different libraries.
+  module Middleware
+    # Object yielded by VCR's {NetHttp2} middleware that allows you to configure
+    # the cassette dynamically based on the rack env.
+    class CassetteArguments
```

Comment:
question: This part looks duplicated from lib/vcr/middleware/rack.rb maybe we can extract something to avoid duplication?